File: Aplikacija/BE/DiplomskiRAD/DTOs/ServiceDTO.cs
using DiplomskiRAD.Enums;
using DiplomskiRAD.Models;
using System.ComponentModel.DataAnnotations.Schema;
using static DiplomskiRAD.DTOs.UserDTO;

namespace DiplomskiRAD.DTOs
{
    public class ServiceDTO
    {
       
        public class UserServiceInfo
        {
            public Guid Id { get; set; }

            public string FailureDescription { get; set; }

            public string? Picture { get; set; }

            public string StartDate { get; set; } //datum pospeca zahteva 

            public string? EndDate { get; set; }

            public ServiceStatus ServiceStatus { get; set; }

            public string? razlogOdbijanja { get; set; }

        }


        public class DirektorServiceInfo
        {
            public Guid Id { get; set; }

            public string FailureDescription { get; set; }

            public string? Picture { get; set; }

            public string StartDate { get; set; } //datum pospeca zahteva 

            public string? EndDate { get; set; }

            public ServiceStatus ServiceStatus { get; set; }

            public string? razlogOdbijanja { get; set; }

            public UserInfo UserInfo { get; set; }

        }

        public class CreateServiceDto
        {
            public string? Picture { get; set; }
            public string FailureDescription { get; set; }

            public string StartDate { get; set; } //datum pospeca zahteva-tj ovde kada ga je poslao

        }

        public class DeleteServiceDto
        {
            public string razlogOdbijanja { get; set; }

        }
    }
}

File: Aplikacija/FE/Diplomski/src/app/models/serviceDTO.ts
import { Review, ReviewInfo } from "./reviewDTO"
import { TaskService, TaskServiceInfo } from "./taskDTO"
import { User, UserInfo } from "./userDTO"

export interface Service
{
    id : string,
    failureDescription : string,
    picture? : string
    startDate : string,
    endDate : string,
    serviceStatus : string,
    razlogOdbijanja : string,
    users : User,
    taskServices:TaskService[]
    reviews : Review[]
}

export interface ServiceInfo
{
    id : string,
    failureDescription : string,
    picture? : string
    startDate : string,
    endDate : string,
    serviceStatus : string,
    razlogOdbijanja : string,
    userInfo: UserInfo,
    taskServiceInfo : TaskServiceInfo[]
    reviewInfos : ReviewInfo[]
}


export interface UserServiceInfo 
{
    id : string,
    failureDescription : string,
    picture? : string
    startDate : string,
    endDate : string,
    serviceStatus : string,
    razlogOdbijanja : string,
    reviewInfos : ReviewInfo[]
}

export interface DirektorServiceInfo
{
    id : string,
    failureDescription : string,
    picture? : string
    startDate : string,
    endDate : string,
    serviceStatus : string,
    razlogOdbijanja : string,
    userInfo: UserInfo,
    taskServiceInfo : TaskServiceInfo[],
    reviewInfos : ReviewInfo[],
}

export interface CreateServiceDto
{
    failureDescription : string,
    picture? : string
    startDate : string,
}

export interface DeclineServiceDto
{
    razlogOdbijanja : string
}